src/dev/colorgenPlugin.ts

```diff
@@ -1,145 +1,34 @@
-      ? {
-          base: 40,
-          onBase: 100,
-          baseContainer: 90,
-          onBaseContainer: 10,
-        }
-      : {
-          base: 80,
-          onBase: 20,
-          baseContainer: 30,
-          onBaseContainer: 90,
-        };
-
-  const getColor = (color: string, palette: TonalPalette) => {
-    color = pascalToKebab(color);
-
-    return [
-      [color, palette.tone(tones.base)],
-      [`on-${color}`, palette.tone(tones.onBase)],
-      [`${color}-container`, palette.tone(tones.baseContainer)],
-      [`on-${color}-container`, palette.tone(tones.onBaseContainer)],
-    ];
-  };
-
-  const results: Record<keyof QuaffColors, number> = Object.fromEntries(
-    Object.entries(palettes)
-      .map(([color, palette]) => getColor(color, palette))
-      .flat(1)
-  );
-
-  results["surface"] = palettes.neutral.tone(mode === "light" ? 98 : 6);
-  results["surface-dim"] = palettes.neutral.tone(mode === "light" ? 87 : 6);
-  results["surface-bright"] = palettes.neutral.tone(mode === "light" ? 98 : 24);
-
-  results["on-surface"] = palettes.neutral.tone(mode === "light" ? 10 : 90);
-  results["on-surface-variant"] = palettes.neutralVariant.tone(mode === "light" ? 30 : 90);
-
-  results["surface-container-lowest"] = palettes.neutral.tone(mode === "light" ? 100 : 4);
-  results["surface-container-low"] = palettes.neutral.tone(mode === "light" ? 96 : 10);
-  results["surface-container"] = palettes.neutral.tone(mode === "light" ? 94 : 12);
-  results["surface-container-high"] = palettes.neutral.tone(mode === "light" ? 92 : 17);
-  results["surface-container-highest"] = palettes.neutral.tone(mode === "light" ? 90 : 24);
-
-  results["inverse-surface"] = palettes.neutral.tone(mode === "light" ? 20 : 90);
-  results["inverse-on-surface"] = palettes.neutral.tone(mode === "light" ? 95 : 20);
-  results["inverse-primary"] = palettes.primary.tone(mode === "light" ? 80 : 40);
-
-  results["outline"] = palettes.neutralVariant.tone(mode === "light" ? 50 : 60);
-  results["outline-variant"] = palettes.neutralVariant.tone(mode === "light" ? 80 : 30);
-
-  results["scrim"] = palettes.neutral.tone(0);
-  results["shadow"] = palettes.neutral.tone(0);
-
-  return Object.fromEntries(
-    Object.entries(results).map(([color, value]) => [color, hexFromArgb(value)])
-  ) as QuaffColors;
-}
-
-function stringFromPalette(palette: QuaffColors) {
+function stringFromPalette(palette: QuaffColors, mode: string, type: "root" | "body") {
+  const modeString = mode ? `-${mode}` : "";
   return Object.entries(palette)
-    .map(([color, hex]) => `--${color}: ${hex};`)
+    .map(([color, hex]) =>
+      type === "root" ? `--${color}${modeString}: ${hex};` : `--${color}: var(--${color}-${mode});`
+    )
     .join("");
 }
 
-function writeColorFile() {
-  const colors = generateColors();
+export function writeColorFile() {
+  const colors = generateColors(BASE_COLOR);
+  const disclaimer = "// AUTO GENERATED FILE - DO NOT MODIFY OR DELETE";
```

Comment:
Maybe we should have this string in a reusable module, so it will always be the same across every generated file.

Reply:
Agreed

Reply:
8a07a822412711724add5dca3547f144a898a8d0